pkg/virt-handler/device-manager/generic_device.go

```diff
@@ -69,6 +69,15 @@ type GenericDevicePlugin struct {
 	deregistered chan struct{}
 }
 
+type DevicePlugin interface {
+	Start(chan<- struct{}) error
+	ListAndWatch(*pluginapi.Empty, pluginapi.DevicePlugin_ListAndWatchServer) error
+	HealthCheck() error
+	StopDevicePlugin() error
+	GetDeviceName() string
+	GetInitialized() bool
+}
```

Comment:
In my opinion, in order to you not have to rework your patches too much while you learn your way around the code base, I suggest for you to start small. Just the methods related to refactor the initialization, for example.